addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/discovery/NeatoAccountDiscoveryService.java

```diff
@@ -0,0 +1,202 @@
+ */
+public class NeatoAccountDiscoveryService extends AbstractDiscoveryService {
+
+    private static final Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private static final int TIMEOUT = 15;
+    private SecureRandom random = new SecureRandom();
+
+    private String accessToken;
+
+    public NeatoAccountDiscoveryService() throws IllegalArgumentException {
+        super(NeatoHandlerFactory.SUPPORTED_THING_TYPES_UIDS, TIMEOUT);
+        this.accessToken = null;
+    }
+
+    private String sendAuthRequestToNeato(String data) {
+
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/vnd.neato.nucleo.v1");
+
+        if (this.accessToken != null) {
+            headers.setProperty("Token token", this.accessToken);
+        }
+
+        String resultString = "";
+
+        try {
+
+            InputStream stream = new ByteArrayInputStream(data.getBytes(StandardCharsets.UTF_8));
+
+            resultString = HttpUtil.executeUrl("POST", "https://beehive.neatocloud.com/sessions", headers, stream,
+                    "application/json", 20000);
+
+            logger.info(resultString);
+
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Error when sending Authentication request to Neato. Error: {}", e.getMessage());
+
+        } catch (IOException e) {
+            logger.error("Error when sending Authentication request to Neato. Error: {}", e.getMessage());
+
+        }
+        return resultString;
+    }
+
+    private Boolean authenticate(String username, String password) {
+
+        String authenticationString;
+        try {
+            authenticationString = "{\"email\": \"" + username + "\", \"password\": \"" + password
+                    + "\", \"os\": \"ios\", \"token\": \"" + new BigInteger(130, random).toString(64).getBytes("UTF-8")
+                    + "\"}";
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Error when during Authentication procedure. Error: {}", e.getMessage());
+
+            return false;
+        }
+
+        String authenticationResponse = sendAuthRequestToNeato(authenticationString);
+
+        logger.info("Authentication Response: {}", authenticationResponse);
+
+        Gson gson = new Gson();
+
+        BeehiveAuthentication authenticationObject = gson.fromJson(authenticationResponse, BeehiveAuthentication.class);
+        this.accessToken = authenticationObject.getAccessToken();
+
+        return true;
+    }
+
+    public Boolean sendGetRobots() {
+
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/vnd.neato.nucleo.v1");
+
+        if (this.accessToken != null) {
+            headers.setProperty("Authorization", "Token token=" + this.accessToken);
+        }
+
+        String resultString = "";
+
+        try {
+
+            resultString = HttpUtil.executeUrl("GET", "https://beehive.neatocloud.com/dashboard", headers, null,
```

Comment:
Move the declaration of the `resultString` in here, you don't need it outside the try-catch